services/cloudfront.py

```diff
@@ -2,40 +2,137 @@
+        distribution = client.get_distribution(Id=distribution["Id"])["Distribution"]
+
+        if "WebACLId" in distribution["DistributionConfig"] and distribution["DistributionConfig"]["WebACLId"] != "":
+            compliant_resources.append(distribution["ARN"])
+        else:
+            non_compliant_resources.append(distribution["ARN"])
+
     return RuleCheckResult(
-        passed=False, compliant_resources=[], non_compliant_resources=[]
+        passed=not non_compliant_resources,
+        compliant_resources=compliant_resources,
+        non_compliant_resources=non_compliant_resources,
     )
 
 
 def cloudfront_default_root_object_configured():
+    compliant_resources = []
+    non_compliant_resources = []
+    distributions = client.list_distributions()["DistributionList"]["Items"]
+
+    for distribution in distributions:
+        distribution = client.get_distribution(Id=distribution["Id"])["Distribution"]
+
+        if distribution["DistributionConfig"]["DefaultRootObject"] != "":
+            compliant_resources.append(distribution["ARN"])
+        else:
+            non_compliant_resources.append(distribution["ARN"])
+
     return RuleCheckResult(
-        passed=False, compliant_resources=[], non_compliant_resources=[]
+        passed=not non_compliant_resources,
+        compliant_resources=compliant_resources,
+        non_compliant_resources=non_compliant_resources,
     )
 
 
 def cloudfront_no_deprecated_ssl_protocols():
+    compliant_resources = []
+    non_compliant_resources = []
+    distributions = client.list_distributions()["DistributionList"]["Items"]
+
+    for distribution in distributions:
+        distribution = client.get_distribution(Id=distribution["Id"])["Distribution"]
+
+        for origin in distribution["DistributionConfig"]["Origins"]["Items"]:
+            if (
+                "CustomOriginConfig" in origin
+                and "SSLv3" in origin["CustomOriginConfig"]["OriginSslProtocols"]["Items"]
+            ):
+
+                non_compliant_resources.append(distribution["ARN"])
+                break
+        else:
+            compliant_resources.append(distribution["ARN"])
+
     return RuleCheckResult(
-        passed=False, compliant_resources=[], non_compliant_resources=[]
+        passed=not non_compliant_resources,
+        compliant_resources=compliant_resources,
+        non_compliant_resources=non_compliant_resources,
     )
 
 
 def cloudfront_s3_origin_access_control_enabled():
+    compliant_resources = []
+    non_compliant_resources = []
+    distributions = client.list_distributions()["DistributionList"]
+
+    for distribution in distributions["Items"]:
+        for origin in distribution["Origins"]["Items"]:
+            if "S3OriginConfig" in origin and origin["OriginAccessControlId"] == "":
+                non_compliant_resources.append(distribution["Id"])
+                break
+        else:
+            compliant_resources.append(distribution["Id"])
```

Comment:
`distribution["Id"]` 대신 `distribution["ARN"]`을 사용합니다.
```suggestion
                non_compliant_resources.append(distribution["ARN"])
                break
        else:
            compliant_resources.append(distribution["ARN"])
```